pkg/config/config.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/amazon-eks-pod-identity-webhook/pkg/filesystem"
+	"k8s.io/klog/v2"
+	"sync"
+)
+
+type Config interface {
+	Get(namespace string, serviceAccount string) *ContainerCredentialsPatchConfig
+}
+
+type FileConfig struct {
+	containerCredentialsAudience string
+	containersCredentialsFullUri string
+
+	watcher              *filesystem.FileWatcher
+	identityConfigObject *IdentityConfigObject
+	cache                map[Identity]bool
+	mu                   sync.RWMutex // guards cache
```

Comment:
I would store this as a `ptr`, given that if somebody decides to "copy" the `FileConfig` object, the copy will retain a copy to the same cache, watcher and `identityConfigObject` but it will have an entirely different mutex associated with it.

Reply:
I will keep it as is.  Looking at the k8s codebase and the webhook code base, most structs use a mixture of ptr & non-ptr.  For Mutex, most golang examples dont declare it as a pointer.